core/app/models/spree/product.rb

```diff
@@ -32,12 +33,15 @@ class Product < Spree::Base
       include Spree::VendorConcern
     end
 
-    MEMOIZED_METHODS = %w(total_on_hand taxonomy_ids taxon_and_ancestors category
+    MEMOIZED_METHODS = %w[total_on_hand taxonomy_ids taxon_and_ancestors category
                           default_variant_id tax_category default_variant
-                          purchasable? in_stock? backorderable?)
+                          purchasable? in_stock? backorderable?]
 
-    friendly_id :slug_candidates, use: :history
+    TRANSLATABLE_FIELDS = %i[name description slug meta_description meta_keywords meta_title].freeze
+    translates(*TRANSLATABLE_FIELDS)
+    Product::Translation.class_eval { acts_as_paranoid }
```

Comment:
I can't imagine why it wouldn't work, `class_eval` is basically equivalent to adding a line of code to the class itself, so it should have the same effect as, for example, adding `acts_as_paranoid` to a class of ours that is explicitly defined (rather than the Product::Translation class that is created dynamically and has no set file where the class is defined). Does that make sense?

Reply:
Yes, it does :) My concern was mostly about the way decorators should be defined in Spree to avoid issues with autoloader, but I guess this won't apply since it's inside spree_core. 

Just a question - what triggers the initial definition of the `Product::Translation` class by Mobility? Does it happen the moment we `include TranslatableResource`?

Reply:
I'm not sure what specifically triggers it, I know that Mobility requires you to define which fields are to be translated in order for the Translation class to be create, so it's either at `include TranslatableResource` or at `translate(*TRANSLATABLE_FIELDS)`. 

You can also check out [how the class is dynamically constructed here](https://github.com/shioyama/mobility/blob/afeb067488f215cb469cfc9d7ad0a3853ae8ac5d/lib/mobility/backends/active_record/table.rb#L239), I had needed to step through the code during earlier debugging 😆